File: src/main/java/com/proyecto/apirestfull/service/TransaccionService.java
package com.proyecto.apirestfull.service;

import com.proyecto.apirestfull.model.Transaccion;

import java.util.List;

public interface TransaccionService {
    List<Transaccion> listarTodas();
    Transaccion obtenerPorId(Integer id);
    List<Transaccion> obtenerPorCuenta(Integer cuentaId);
    Transaccion registrar(Transaccion transaccion);
}

File: frontend/src/app/services/transaccion.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Transaccion } from '../models/transaccion.model';

@Injectable({ providedIn: 'root' })
export class TransaccionService {
  private readonly baseUrl = '/api/transacciones';

  constructor(private http: HttpClient) {}

  listar(): Observable<Transaccion[]> {
    return this.http.get<Transaccion[]>(this.baseUrl);
  }

  obtener(id: number): Observable<Transaccion> {
    return this.http.get<Transaccion>(`${this.baseUrl}/${id}`);
  }

  obtenerPorCuenta(cuentaId: number): Observable<Transaccion[]> {
    return this.http.get<Transaccion[]>(`${this.baseUrl}/cuenta/${cuentaId}`);
  }

  registrar(transaccion: Transaccion): Observable<Transaccion> {
    return this.http.post<Transaccion>(this.baseUrl, transaccion);
  }
}
